File: java/src/main/java/ml/wonwoo/springdatashowcase/product/Book.java
package ml.wonwoo.springdatashowcase.product;

import javax.money.MonetaryAmount;
import javax.persistence.Entity;

@Entity
public class Book extends Product {

    public enum BookType {
        SOFTWARE,
        SCIENCE,
        FANTASY,
    }

    private String image;
    private String author;
    private BookType type;

    public Book(String name, String image, String author, MonetaryAmount price, BookType type) {
        super(name, price);
        this.image = image;
        this.author = author;
        this.type = type;
    }

    private Book() {

    }

    public String getImage() {
        return image;
    }

    public BookType getType() {
        return type;
    }

    public String getAuthor() {
        return author;
    }
}

File: kotlin/src/main/kotlin/ml/wonwoo/springdatashowcase/product/Book.kt
package ml.wonwoo.springdatashowcase.product

import javax.money.MonetaryAmount
import javax.persistence.Entity

@Entity
data class Book(

    override val name: String,

    override val price: MonetaryAmount,

    val image: String,

    val author: String,

    val type: BookType

) : Product(name = name, price = price)

enum class BookType {
    SOFTWARE,
    SCIENCE,
    FANTASY
}
